execution/gethexec/executionengine.go

```diff
@@ -1009,3 +1010,7 @@ func (s *ExecutionEngine) Start(ctx_in context.Context) {
 		})
 	}
 }
+
+func (s *ExecutionEngine) Maintenance(capLimit uint64) error {
+	return s.bc.FlushTrieDB(&s.createBlocksMutex, common.StorageSize(capLimit))
```

Comment:
problem is on geth side..
FlushTriedb should not take pointer to a lock.
The function in blockchain should lock chainmu (like other blockchain funcs)
This Maintenance function should probably lock createBlockMutex as well

Reply:
Done.